File: Python/Problems/Problems/problem1779.py
import sys
from typing import List

from .problem import Problem

class Problem1779(Problem):
    def test(self) -> bool:
        x = 3
        y = 4

        points = [[1, 2], [3, 1], [2, 4], [2, 3], [4, 4]]

        expected = 2

        result = self.nearestValidPoint(x, y, points)

        print(result)

        return expected == result

    def nearestValidPoint(self, x: int, y: int, points: List[List[int]]) -> int:
        minDist = sys.maxsize
        result = -1
        for i in range(len(points)):
            if x != points[i][0] and y != points[i][1]:
                continue
            dist = abs(x - points[i][0]) + abs(y - points[i][1])
            if dist < minDist:
                result = i
                minDist = dist

        return result

File: C++/Problems/Problems/problem_1779.cpp
#include "problem_1779.h"

bool problem_1779::test() {
    auto x = 3;
    auto y = 4;

    std::vector<std::vector<int>> points { { 1, 2 }, { 3, 1 }, { 2, 4 }, { 2, 3 }, { 4, 4 } };

    auto expected = 2;

    auto result = nearestValidPoint(x, y, points);

    std::cout << result << std::endl;

    return expected == result;
}

int problem_1779::nearestValidPoint(int x, int y, std::vector<std::vector<int>> &points) {
    auto minDist = INT_MAX;
    auto result = -1;
    for (int i = 0; i < points.size(); i++)
    {
        if (x != points[i][0] && y != points[i][1])
            continue;
        auto dist = std::abs(x - points[i][0]) + std::abs(y - points[i][1]);
        if (dist < minDist)
        {
            result = i;
            minDist = dist;
        }
    }

    return result;
}
